MobiFlight-Installer/MobiFlight-Installer/UI/InstallMainForm.cs

```diff
@@ -0,0 +1,156 @@
+                Log.Instance.log("DOWNLOAD is incorrect, installation aborted", LogSeverity.Debug);
+                MessageBox.Show("Download failed, installation aborted! Please run installer again or unzip manually.");
+            }
+            _webClient.Dispose();
+        }
+
+        private void ManualUpgradeFromCommandLine(string Version)
+        {
+            if (MobiFlightUpdaterModel.resultList[Version]["url"].Length > 5)
+            {
+                _downloadURL = MobiFlightUpdaterModel.resultList[Version]["url"];
+                _downloadChecksum = MobiFlightUpdaterModel.resultList[Version]["checksum"];
+                CurrentFileName = PathToInstall + "\\" + MobiFlightUpdaterModel.GetFileName(_downloadURL);
+
+                if (!MobiFlightUpdaterModel.CheckIfFileIsHere(CurrentFileName, _downloadChecksum)) //compare checksum if download the file is needeed
+                {
+
+                    SetupProgressBar.Value = 0;
+                    UpdaterCurrentTask.Text = "Downloading...";
+                    Log.Instance.log("Downloading " + CurrentFileName, LogSeverity.Debug);
+                    _webClient = new WebClient();
+                    var uri = new Uri(_downloadURL);
+                    _webClient.DownloadProgressChanged += new DownloadProgressChangedEventHandler(OnDownloadProgressChanged);
+                    _webClient.DownloadFileCompleted += new AsyncCompletedEventHandler(OnDownloadComplete);
+                    _webClient.DownloadFileAsync(uri, CurrentFileName); // Download the file
+                }
+            }
+            else
+            {
+                Log.Instance.log("URL is incorrect, installation aborted", LogSeverity.Debug);
+                MessageBox.Show("URL is incorrect, impossible to download the file, installation aborted!");
+            }
+        }
+
+        private async void AsyncCheckGithub()
+        {
+            await CheckGithub();
+        }
+        private async Task CheckGithub()
+        {
+            try
+            {
+                using (var client = new HttpClient())
+                {
+                    await client.GetAsync("https://github.com");
+                    BoxGithub.Checked = true;
+                    IfReadyForTakeOFF();
+                }
+            }
+            catch
+            {
+                BoxGithub.Checked = false;
+                IfReadyForTakeOFF();
+            }
+        }
+
+        private void IfReadyForTakeOFF()
+        {
+            if (BoxGithub.Checked & BoxTargetDirectory.Checked)
+            {
+                SetupTitle.Text = "Clear to take off MF9";
+                SetupTitle.ForeColor = System.Drawing.Color.White;
+            }
+            else
+            {
+                SetupTitle.Text = "Take OFF aborted, something wrong";
+                SetupTitle.ForeColor = System.Drawing.Color.Red;
+            }
+        }
+
+        private void label1_Click(object sender, EventArgs e)
```

Comment:
This was in the original installer, but should be removed anyway since it doesn't do anything.